src/app/loadout-builder/utils.ts

```diff
@@ -210,82 +139,25 @@ export function getFilteredPerksAndPlugSets(
   return { filteredPlugSetHashes, filteredPerks };
 }
 
-/** Whether this item is eligible for being in loadout builder */
+/** Whether this item is eligible for being in loadout builder. Now only armour 2.0 and only items that have all the stats. */
 export function isLoadoutBuilderItem(item: DimItem) {
   // Armor and Ghosts
-  return item.bucket.inArmor;
+  return (
+    item.bucket.inArmor &&
+    item.energy &&
+    statValues.every((statHash) => item.stats?.some((dimStat) => dimStat.statHash === statHash))
+  );
 }
 
 export function statTier(stat: number) {
   return _.clamp(Math.floor(stat / 10), 0, 10);
 }
 
 /**
- * This figures out which perks need to be selected on specific armor 1.0 items to achieve the desired stat mix.
- * process#generateMixesFromPerksOrStats uses a very similar algorithm to generate the stat mixes initially so the
- * two should be kept in sync if this changes.
- */
-export function generateMixesFromPerks(
```

Comment:
Love the cleanup. I wonder if there's even more we can remove now that each item has one and only one stat set instead of having N possible mixes.

Reply:
I think there is, I will go digging for more.

Reply:
I think we were generating all the stat sets before but only keeping the "best" one which why you don't see much of it getting removed here. We were recalculating it if needed in the UI. The stats on ArmorSet are now just a record of stat type to a number.